src/main/java/uk/gov/companieshouse/filinghistory/consumer/delta/TransactionKindService.java

```diff
@@ -0,0 +1,30 @@
+package uk.gov.companieshouse.filinghistory.consumer.delta;
+
+import static org.apache.commons.lang3.StringUtils.trim;
+
+import java.nio.charset.StandardCharsets;
+import org.apache.commons.codec.binary.Base64;
+import org.apache.commons.lang3.StringUtils;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+import uk.gov.companieshouse.api.filinghistory.InternalData.TransactionKindEnum;
+
+@Component
+public class TransactionKindService {
+
+    private final String transactionIdSalt;
+
+    public TransactionKindService(@Value("${transaction-id-salt}") String transactionIdSalt) {
+        this.transactionIdSalt = transactionIdSalt;
+    }
+
+    public TransactionKindResult encodeIdByTransactionKind(TransactionKindCriteria transactionKindCriteria) {
+        return new TransactionKindResult(encodeTransactionId(transactionKindCriteria.entityId()),
```

Comment:
We should add the encoded transactionId to the DataLogMap and log a debug message here including the transaction Kind.